contracts/actions/mcd/McdBoostComposite.sol

```diff
@@ -0,0 +1,248 @@
+ReentrancyGuard, MainnetBalancerV2Addresses {
+    using TokenUtils for address;
+
+    address internal immutable ACTION_ADDR = address(this);
+
+    /// @param _vaultId Id of the vault
+    /// @param _amount Amount of dai to be payed back
+    /// @param _from Where the Dai is pulled from
+    /// @param _mcdManager The manager address we are using
+    struct BoostParams {
+        uint256 vaultId;
+        uint256 boostAmount;
+        address mcdManager;
+        address joinAddr;
+        ExchangeData exchangeData;
+    }
+
+    function executeAction(
+        bytes calldata _callData,
+        bytes32[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override (ActionBase, DFSSell) returns (bytes32) {
+        BoostParams memory boostParams = _parseCompositParams(_callData);
+
+        boostParams.vaultId = _parseParamUint(
+            boostParams.vaultId,
+            _paramMapping[0],
+            _subData,
+            _returnValues
+        );
+
+        boostParams.boostAmount = _parseParamUint(
+            boostParams.boostAmount,
+            _paramMapping[1],
+            _subData,
+            _returnValues
+        );
+
+        boostParams.mcdManager = _parseParamAddr(
+            boostParams.mcdManager,
+            _paramMapping[2],
+            _subData,
+            _returnValues
+        );
+
+        boostParams.joinAddr = _parseParamAddr(
+            boostParams.joinAddr,
+            _paramMapping[3],
+            _subData,
+            _returnValues
+        );
+
+        boostParams.exchangeData.srcAddr = _parseParamAddr(
+            boostParams.exchangeData.srcAddr,
+            _paramMapping[4],
+            _subData,
+            _returnValues
+        );
+        boostParams.exchangeData.destAddr = _parseParamAddr(
+            boostParams.exchangeData.destAddr,
+            _paramMapping[5],
+            _subData,
+            _returnValues
+        );
+
+        boostParams.exchangeData.srcAmount = _parseParamUint(
+            boostParams.exchangeData.srcAmount,
+            _paramMapping[6],
+            _subData,
+            _returnValues
+        );
+
+        _flBalancer(boostParams);
+    }
+
+    /// @notice Parses inputs and runs the single implemented action through a proxy
+    /// @dev Used to save gas when executing a single action directly
+    function executeActionDirect(bytes memory _callData) public payable virtual override (ActionBase, DFSSell) {
+        BoostParams memory boostParams = _parseCompositParams(_callData);
+        _flBalancer(boostParams);
+    }
+
+    /// @notice Gets a FL from Balancer and returns back the execution to the action address
+    function _flBalancer(BoostParams memory _boostParams) internal {
+        address[] memory tokens = new address[](1);
+        uint256[] memory amounts = new uint256[](1);
+        tokens[0] = _boostParams.exchangeData.srcAddr;
```

Comment:
can be hardcoded to DAI? or have a check if .srcAddr is not DAI to revert on start to not spend more gas than needed later on

Reply:
makes sense